### ISSUE TITLE
javascript tests are failing

### ISSUE THREAD

**Issue by juho-jaakkola** [Elgg/Elgg#9139]
```npm test``` is failing:

```
WARN [watcher]: Pattern "/home/travis/build/Elgg/Elgg/vendor/bower-asset/jquery/dist/jquery.js" does not match any file.
INFO [karma]: Karma v0.12.37 server started at http://localhost:9876/
INFO [launcher]: Starting browser PhantomJS
INFO [PhantomJS 1.9.8 (Linux 0.0.0)]: Connected on socket ZDQwCMgNp-NU49JyYxb0 with id 55184644
PhantomJS 1.9.8 (Linux 0.0.0) ERROR
ReferenceError: Can't find variable: jQuery
at /home/travis/build/Elgg/Elgg/js/lib/elgglib.js:42
```
https://travis-ci.org/Elgg/Elgg/jobs/92918942#L607

**Comment by juho-jaakkola** [comment]
I haven't been able to reproduce the error locally.

**Comment by juho-jaakkola** [comment]
@mrclay said:
> We can give Travis a user token, but anyone could create PR that exposes it. The token wouldn't need any special scopes so maybe that's OK? http://blog.wyrihaximus.net/2015/09/github-auth-token-on-travis/

Sounds good to me.

**Comment by mrclay** [comment]
And as I read later, the token holder can read from private repos the user has access to. So we'd want to make a GitHub user with no formal association just for Travis.

Which has me wondering: would using bower directly have the same problems on Travis?

**Comment by mrclay** [comment]
See https://github.com/francoispluchino/composer-asset-plugin/issues/168#issuecomment-159531055

**Comment by mrclay** [comment]
Fixed by #9145

**Comment by mrclay** [comment]
Nope. Despite https://github.com/mrclay/Elgg-leaf/commit/d05bc7e63d2bff82e8094499e82de3a138a89117 assets fail to install randomly. I think we could [specify the full asset versions](https://github.com/francoispluchino/composer-asset-plugin/blob/master/Resources/doc/index.md#reduce-the-number-of-requests-for-getting-the-package-definitions) to reduce API requests.
